tests/tests_pytorch/accelerators/test_mps.py

```diff
@@ -162,3 +162,14 @@ def to(self, *args, **kwargs):
 
     assert batch.text.type() == "torch.mps.LongTensor"
     assert batch.label.type() == "torch.mps.LongTensor"
+
+
+@RunIf(mps=True)
+def test_data_is_not_changed_after_move_to_mps_device():
+    trainer = Trainer(accelerator="mps", devices=1)
+    x = torch.zeros([10, 10])
+    device = torch.device("mps")
+
+    for _ in range(1000):
```

Comment:
I'm hesitant of the robustness of this test. It might be better to not add a test at all actually.

Reply:
Yeah you probably right. Creating a test case for a race bug like this is rather hard and will always be flaky. I can create a test case to verify that `.to` is called without `non_blocking`. However this will require some mocking or patching. So I guess that is the reason this test case does not exist for the `cpu` case. So I guess you are right. It might be better to not add a test case for this.